src/code/audio/code_8019AF00.c

```diff
@@ -2047,80 +3060,388 @@ void AudioOcarina_TerminaWallGenerateNotes(void) {
+
+        if ((sPlaybackState == 0) && (sIsOcarinaInputEnabled == true)) {
+            AudioOcarina_PlayControllerInput(false);
+        }
+
+        if (sOcarinaFlags != 0) {
+            if (sOcarinaFlags & 0x40000000) {
+                AudioOcarina_CheckSongsWithoutMusicStaff();
+            } else {
+                AudioOcarina_CheckSongsWithMusicStaff();
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D600.s")
+        AudioOcarina_PlaybackSong();
+        sOcarinaPlaybackTaskStart = sOcarinaUpdateTaskStart;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D758.s")
+        if (sPlaybackState == 0) {
+            AudioOcarina_RecordSong();
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D864.s")
+        if ((sOcarinaFlags != 0) && (sPrevOcarinaPitch != sCurOcarinaPitch)) {
+            sOcarinaDropInputTimer = 1; // Drops ocarina input for 1 frame
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D8B4.s")
+        sPrevOcarinaPitch = sCurOcarinaPitch;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D8E4.s")
+        if (sOcarinaDisableTimer != 0) {
+            sOcarinaDisableTimer--;
+            if (sOcarinaDisableTimer == 0) {
+                AudioOcarina_SetInstrument(OCARINA_INSTRUMENT_OFF);
+            }
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019DF28.s")
+    AudioOcarina_UpdatePlayingStaff();
+    AudioOcarina_UpdatePlaybackStaff();
+    AudioOcarina_UpdateRecordingStaff();
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019DF64.s")
+void AudioOcarina_PlayLongScarecrowAfterCredits(void) {
+    static u8 sScarecrowAfterCreditsState = 0;
+    static u8 sScarecrowAfterCreditsIntrumentId = OCARINA_INSTRUMENT_DEFAULT;
+    static u16 sScarecrowAfterCreditsTimer = 1200;
+
+    switch (sScarecrowAfterCreditsState) {
+        case 0:
+            if (sScarecrowAfterCreditsTimer-- == 0) {
+                // OoT originally had 7 Ocarina instruments
+                // MM has more so this does not go all the way to the max
+                if (sScarecrowAfterCreditsIntrumentId < 7) {
+                    // set next ocarina instrument and restart
+                    sScarecrowAfterCreditsState++;
+                } else {
+                    // finished
+                    sScarecrowAfterCreditsState = 3;
+                    AudioOcarina_SetInstrument(OCARINA_INSTRUMENT_OFF);
+                }
+                sScarecrowAfterCreditsTimer = 1200;
+            }
+            break;
+
+        case 1:
+            Audio_SetSfxBanksMute(0);
+            AudioOcarina_SetInstrument(sScarecrowAfterCreditsIntrumentId);
+            AudioOcarina_SetPlaybackSong(OCARINA_SONG_SCARECROW_LONG + 1, 1);
+            sScarecrowAfterCreditsIntrumentId++;
+            sScarecrowAfterCreditsState++;
+            break;
+
+        case 2:
```

Comment:
(Agreed that it's not really worth defines with the way this state is set and used.)